### ISSUE TITLE
Snippy wrapper fails with Collection of Paired dataset 

### ISSUE THREAD

**Issue by jowong4** [galaxyproject/tools-iuc#2161]
#set $dir_name = re.sub('[^\w_]', '_', $fastq_input.fastq_input1.element_identifier) this line of code will fail when the input is  collection of paired datasets

**Comment by dfornika** [comment]
@jowong4 Could you provide a bit more context?

I think this is the line you're referring to:

https://github.com/galaxyproject/tools-iuc/blob/57493fdc88e6128bed127993464f2a6480e2774d/tools/snippy/snippy.xml#L60

Do you get an error message? What do you think is going wrong?

**Comment by jowong4** [comment]
The problem with that line is it doesnt work with collection of paired dataset. The data structure is different and the element_identifier line which normally can work with collection and single data file will fail. When you run that script with a collection of paired datasets, the job will run into a stop and return "elemt_identifier" as an error message.


The author's intention behind that line of code is to make a folder with the name of the sample. As of now, I do not know how to do that directly. I can obtain it indirectly through running a throwaway function, but I have no way of obtaining it within the same script at the moment of taking an collection of paired dataset.

**Comment by dfornika** [comment]
@Slugger70 @tseemann Do either of you have a chance to look at this issue? Is there a way to run `snippy` on a collection of paired datasets?

**Comment by jowong4** [comment]
@dfornika  It isn't so much that snippy wouldnt run with collection of paired datasets, but that the steps to set up for snippy-core breaks the entire wrapper. If you remove the code that sets up the input for snippy-core, the wrapper works fine

**Comment by tseemann** [comment]
@dfornika I think @Slugger70 had to do a LOT of hacking to get things working, as Galaxy doesn't really support output "folders".

@Slugger70 snippy >= 4.3 now comes with a `snippy-multi` script for running bulk jobs and automaticcaly running `snippy-core` at the end BUT it ill obviously not parallelize across galaxy nodes.

**Comment by Slugger70** [comment]
It'll require a rethink. I'll try and find some time soon.

**Comment by dfornika** [comment]
I'm taking a crack at updating the snippy tools in this repo to v4.3.6. I'm working in this branch:

https://github.com/dfornika/tools-iuc/tree/snippy-4.3.6

**Comment by dfornika** [comment]
I found some relevant documentation so I'm just posting it here for future reference:

https://planemo.readthedocs.io/en/latest/writing_advanced.html#multiple-output-files

**Comment by dfornika** [comment]
I've submitted a work-in-progress pull request: #2252

**Comment by dfornika** [comment]
Ok, I think I've mostly got this sorted out. I adjusted the `fastq_input` parameter a bit:

```xml
<conditional name="fastq_input">
    <param name="fastq_input_selector" type="select" label="Single or Paired-end reads" help="Select between paired and single end data">
        <option value="paired">Paired</option>
        <option value="single">Single</option>
        <option value="paired_collection">Paired Collection</option>
        <option value="paired_iv">Paired Interleaved</option>
    </param>
    <when value="paired">
        <param name="fastq_input1" type="data" format="fastqsanger,fasta" label="Select first set of reads" help="Specify dataset with forward reads"/>
        <param name="fastq_input2" type="data" format="fastqsanger,fasta" label="Select second set of reads" help="Specify dataset with reverse reads"/>
    </when>
    <when value="single">
        <param name="fastq_input" type="data" format="fastqsanger,fasta" label="Select fastq dataset" help="Specify dataset with single reads"/>
    </when>
    <when value="paired_collection">
        <param name="fastq_input" format="fastqsanger,fasta" type="data_collection" collection_type="paired" label="Select a paired collection" help="See help section for an explanation of dataset collections"/>
    </when>
    <when value="paired_iv">
        <param name="fastq_input" type="data" format="fastqsanger" label="Select fastq dataset" help="Specify dataset with interleaved reads"/>
    </when>
</conditional>
```

...which simplifies the `element_identifier` logic here:

````
#import re
#if str( $fastq_input.fastq_input_selector ) == "paired"
    #set $dir_name = re.sub('[^\w_]', '_', $fastq_input.fastq_input1.element_identifier)
#else
    #set $dir_name = re.sub('[^\w_]', '_', $fastq_input.fastq_input.element_identifier)
#end if

mkdir -p ${dir_name} && cp -r out/reference out/snps.tab out/snps.aligned.fa out/snps.vcf ${dir_name}/ &&
tar -czf out.tgz ${dir_name}
```

**Comment by dfornika** [comment]
I think I've got the Paired Dataset Collection issue sorted out, but the `snippy-4.3.6` tool is going to take a bit more work.

I'm going to submit a separate pull request to simply fix this issue in the existing `snippy-3.2` tool.
